sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -154,20 +154,30 @@ AZ_NODISCARD az_result az_iot_hub_client_sas_signature_get(
  * @param[in] client The #az_iot_hub_client to use for this call.
  * @param[in] base64_hmac_sha256_signature The Base64 encoded value of the HMAC-SHA256(signature,
  *                                         SharedAccessKey). The signature is obtained by using
- *                                         #az_iot_hub_client_sas_signature_get.
+ *                                         #az_iot_hub_client_sas_get_signature.
+ * @param[in] token_expiration_epoch_time The time, in seconds, from 1/1/1970.
+ *                                        It MUST be the same value passed to
+ *                                        az_iot_hub_client_sas_get_signature.
  * @param[in] key_name The Shared Access Key Name (Policy Name). This is optional. For security
  *                     reasons we recommend using one key per device instead of using a global
  *                     policy key.
- * @param[in] mqtt_password An empty #az_span with sufficient capacity to hold the MQTT password.
- * @param[out] out_mqtt_password The output #az_span containing the MQTT password.
+ * @param[out] mqtt_password A char buffer with sufficient capacity to hold the MQTT password.
+ * @param[in] mqtt_password_size The size, in bytes of \p mqtt_password.
+ * @param[out] out_mqtt_password_length __[nullable]__ Contains the string length, in bytes, of
+ *                                                     \p mqtt_password. Can be `NULL`.
  * @return #az_result.
+ *         #AZ_OK if successful. In this case, `mqtt_password` will contain a null-terminated string

```

Comment:
This won't render well in doxygen. Use a list by starting each line with `-`.

Also, use `\p` instead of back ticks when referring to parameter names.